File: javascript/command_injection/reportGenerator.js
const { exec, execSync } = require('child_process');

// VULN 1: exec with template literal - PDF report generation
function generatePdfReport(templateName, outputName) {
    exec(
        `wkhtmltopdf /reports/templates/${templateName}.html /reports/output/${outputName}.pdf`,
        (err) => { if (err) throw err; }
    );
}

// VULN 2: execSync with string concatenation - image resize
function resizeProductImage(imagePath, width, height) {
    return execSync(
        'convert ' + imagePath + ' -resize ' + width + 'x' + height + ' ' + imagePath + '_thumb.jpg'
    ).toString();
}

// VULN 3: exec with string concatenation - video transcode
function transcodeVideo(inputFile, outputFormat, bitrate) {
    exec('ffmpeg -i ' + inputFile + ' -b:v ' + bitrate + ' output.' + outputFormat, (err, stdout) => {
        if (err) throw err;
        return stdout;
    });
}

module.exports = { generatePdfReport, resizeProductImage, transcodeVideo };

File: go/command_injection/report_generator.go
package commandinjection

import (
	"fmt"
	"os/exec"
)

// VULN 1: exec.Command sh -c with string concatenation - PDF generation
func GeneratePDFReport(templateName, outputName string) ([]byte, error) {
	cmdStr := "wkhtmltopdf /reports/templates/" + templateName + ".html /reports/output/" + outputName + ".pdf"
	cmd := exec.Command("sh", "-c", cmdStr)
	return cmd.Output()
}

// VULN 2: exec.Command sh -c with fmt.Sprintf - image resize
func ResizeProductImage(imagePath string, width, height int) ([]byte, error) {
	cmdStr := fmt.Sprintf("convert %s -resize %dx%d %s_resized.jpg", imagePath, width, height, imagePath)
	cmd := exec.Command("sh", "-c", cmdStr)
	return cmd.Output()
}

// VULN 3: exec.Command sh -c with string concatenation - ffmpeg transcode
func TranscodeVideo(inputFile, outputFormat, bitrate string) ([]byte, error) {
	cmd := exec.Command("sh", "-c", "ffmpeg -i "+inputFile+" -b:v "+bitrate+" output."+outputFormat)
	return cmd.Output()
}
